test/fake_http_server.h

```diff
@@ -26,15 +26,40 @@
 namespace google {
 namespace testing {
 
+using GetHandler =
+    // (path, headers) -> body
+    std::function<std::string(const std::string&,
+                              std::map<std::string, std::string>&)>;
+using PostHandler =
+    // (path, headers, body) -> body
+    std::function<std::string(const std::string&,
+                              std::map<std::string, std::string>&,
+                              std::string)>;
+
 // Starts a server in a separate thread, allowing it to choose an
 // available port.
 class FakeServer {
  public:
   FakeServer();
   ~FakeServer();
 
+  // A handler for GET requests that returns a constant string.
+  static GetHandler Return(const std::string& body);
+
+  // Returns the URL for this server without a trailing slash:
+  // http://<host>:<port>
```

Comment:
Want to put this in quotes ("`"http://<host>:<port>"`") and add a trailing `.`?

Reply:
Done.